test/test.ts

```diff
@@ -427,3 +427,70 @@ test('can open a specific channel twice', async t => {
   const ch2 = await conn.channel(2)
   t.is(ch2, ch)
 })
+
+
+test('set basic flow on channel', async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  await t.notThrowsAsync(async () =>  await ch.basicFlow(true))
+})
+
+test('can resolve promise on channel', async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  t.false(ch.resolvePromise())
+})
+
+test('confirming unknown deliveryTag', async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  t.notThrows(() => ch.publishConfirmed(1, false, false))
+})
+
+// ch.deliver does enqueue a microtask, rendering the ch.deliver method untestable.
+test.skip('delivering a message when no consumer exists raises', async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  const msg = new AMQPMessage(ch)
+  msg.consumerTag = "abc"
+  t.throws(() => ch.deliver(msg))
+})
+
+test("can publish as json", async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  t.notThrows(() => ch.basicPublish("amq.topic", "", {test: "hey"}))
+})
+
+test("can publish null", async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  t.notThrows(() => ch.basicPublish("amq.topic", "", null))
+})
+
+test("can publish ArrayBuffer", async t => {
+  const amqp = new AMQPClient("amqp://127.0.0.1")
+  const conn = await amqp.connect()
+  const ch = await conn.channel()
+  t.notThrows(() => ch.basicPublish("amq.topic", "", new ArrayBuffer(2)))
```

Comment:
should try to get the message too, to verify that we get the same data as we published.